### DISCUSSION TITLE
Homeassistant RaspberryMatic RPI-RF-MOD Raspberry Pi 5

### DISCUSSION THREAD

**Discussion by Hannes809** [OpenCCU/OpenCCU#2618] [🇩🇪 Fragen&Antworten] [answered]
Hi ich habe den raspberry pi5. Darauf habe ich Homeassistant und das Addon Raspberrymatic installiert. Nun erkennt Raspberrymatic das Funkmodul nicht. Habe die Config Datei angepasst aber es geht nicht. Mit dem Raspberry pi 4 funktioniert alles ohne Probleme.

```
Mounting /data as /usr/local (Home Assistant Add-On): OK
Starting watchdog...
Identifying host system: Raspberry Pi 5 Model B Rev 1.0 (oci), OK
Initializing RTC Clock: onboard, OK
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Running seedrng: OK
Initializing System: OK
Setup ca-certificates: OK
Starting logging: OK
Init onboard LEDs: init, OK
Starting irqbalance: OK
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.1, OK
Identifying Homematic RF-Hardware: HmRF: n/a, HmIP: n/a, OK
Updating Homematic RF-Hardware: no GPIO/USB connected RF-hardware found
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting ha-proxy: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: not required
Starting rfd: no BidCos-RF hardware found
Starting HMIPServer: ...OK
Starting ReGaHss: .OK
Starting CloudMatic: disabled
Starting Third-Party Addons: OK
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.73.9.20231130 (raspmatic_oci_arm64)
```

**Comment by jens-maus** [comment]
>  Nun erkennt Raspberrymatic das Funkmodul nicht. 

Welches Funkmodul nutzt du denn mit dem RaspberryPi5 und HAos?

**Reply by Hannes809** [reply]
Dieses hier https://de.elv.com/elv-komplettbausatz-funk-modulplatine-fuer-raspberry-pi-3-b-rpi-rf-mod-fuer-homematic-und-homematic-ip-152941 war davor in der CCU3 verbaut

**Comment by Baxxy13** [comment]
War denn schon irgendwo zu lesen das das RPI-RF-MOD auf GPIO in HA-OS (Pi5) "supported" wird?
Wenn ich mir die `config.txt` so ansehe dann fehlt ja der Abschnitt bezüglich RPI-RF-MOD komplett.

Also der hier von den anderen Pi's:
```
# Uncomment this to enable GPIO support for RPI-RF-MOD/HM-MOD-RPI-PCB
#enable_uart=1
#dtparam=i2c_arm=on
#dtoverlay=miniuart-bt
#dtoverlay=rpi-rf-mod
```

Bisher habe ich nur eine unbeantwortete Diskussion zum Thema gefunden:
https://github.com/home-assistant/operating-system/discussions/3069

**Reply by Hannes809** [reply]
Das stimmt hab es einfach hinzugefügt dachte das müsste ausreichen

**Comment by jens-maus** [comment, marked as answer]
Gleiche Diskussion wie hier: https://github.com/home-assistant/operating-system/discussions/3069

Die Essenz scheint zu sein nachdem ich mir das selbst einmal angeschaut habe: Es existiert noch ein Problem in der RaspberryPi5 Version von HomeAssistantOS das dazu führt das die device tree overlays nicht korrekt geladen werden. Zusätzlich zu dem üblichen `dtoverlay` Block:

```
# Uncomment this to enable GPIO support for RPI-RF-MOD/HM-MOD-RPI-PCB
enable_uart=1
dtparam=i2c_arm=on
dtoverlay=miniuart-bt
dtoverlay=rpi-rf-mod
```

zum aktivieren des `RPI-RF-MOD` Supportes muss aktuell noch die folgende zusätzliche Zeile hinzugefügt werden:

```
overlay_prefix=slot-A/overlays/
```

Dann sollte nach einem Reboot das `RPI-RF-MOD` korrekt funktionieren.

**Reply by Hannes809** [reply]
Super vielen dank. Jetzt funktioniert es

**Reply by jens-maus** [reply]
Wunderbar. Hier der komplett-halber der entsprechende Bugreport gegenüber dem HomeAsssistantOS Projekt:

https://github.com/home-assistant/operating-system/issues/3079
